### PR TITLE
allow sorter function to be passed to toArray

### PR DESCRIPTION
**Pull request by GitSparTV**
Completes https://github.com/SinisterRectus/Discordia/projects/2#card-34095190

### CODE REVIEW

File: libs/utils/Bitfield.lua [hunk 1]
```diff
@@ -120,13 +120,16 @@ function Bitfield:__div(other)
...
 
-function Bitfield:toArray(filter)
+function Bitfield:toArray(filter, sort)
 	local arr = {}
 	for k, v in pairs(checkType('table', filter)) do
 		if self:hasValue(v) then
 			insert(arr, k)
 		end
 	end
+	if sort and typing.checkType('table', sort) then
```
Comment (truemedian) [inline libs/utils/Bitfield.lua:130]: `table` is the wrong type here. It must be a `function`.

Comment (GitSparTV) [reply libs/utils/Bitfield.lua:130]: copy-paste mistake, thanks

Comment (Bilal2453) [reply libs/utils/Bitfield.lua:130]: `sort` cannot be false, should be `if sort ~= nil` instead if I'm not mistaken

Comment (truemedian) [reply libs/utils/Bitfield.lua:130]: type checking covers any other case

Comment (GitSparTV) [reply libs/utils/Bitfield.lua:130]: If this is required, alright.

**Comment by truemedian** [comment]
Also note, this is only applied to Bitfield at the moment, eventually it will need to be added to any iterable object.

**Comment by GitSparTV** [comment]
Sinister, should `false` error or not?

**Comment by SinisterRectus** [comment]
The project card is actually for the `Iterable:toArray` method that I have not yet implemented. I'm not sure that I want to include sorters though. I think it's fairly trivial for a user to use `table.sort` on the result if necessary, avoiding the complexity of including it in the method.

**Pull request closed without merging**